pkg/controller/machine/controller.go

```diff
@@ -172,3 +184,19 @@ func (c *MachineControllerImpl) getCluster(machine *clusterv1.Machine) (*cluster
 		return nil, errors.New("multiple clusters defined")
 	}
 }
+
+func (c *MachineControllerImpl) isDeleteAllowed(machine *clusterv1.Machine) bool {
+	if c.nodeName == "" || machine.Status.NodeRef == nil {
+		return true
+	}
+	if machine.Status.NodeRef.Name != c.nodeName {
+		return true
+	}
+	node, err := c.kubernetesClientSet.CoreV1().Nodes().Get(c.nodeName, metav1.GetOptions{})
+	if err != nil {
+		glog.Infof("unable to determine if controller's node is associated with machine '%v', error getting node named '%v': %v", machine.Name, c.nodeName, err)
+		return true
+	}
+	// don't allow this controller to delete its underlying node
```

Comment:
nit: Comments should be full sentences, starting with a capital letter and ending with a period.

Reply:
Fixed.